packages/bsky/src/views/index.ts

```diff
@@ -1606,6 +1595,264 @@ export class Views {
     }
   }
 
+  threadHiddenV2(
+    skeleton: { anchor: string; uris: string[] },
+    state: HydrationState,
+    {
+      below,
+      branchingFactor,
+      viewer,
+    }: {
+      below: number
+      branchingFactor: number
+      viewer: HydrateCtx['viewer']
+    },
+  ): ThreadHiddenItem[] {
+    const { anchor: anchorUri, uris } = skeleton
+
+    // Not found.
+    const postView = this.post(anchorUri, state)
+    const post = state.posts?.get(anchorUri)
+    if (!post || !postView) {
+      return []
+    }
+
+    // Blocked (only 1p for anchor).
+    if (this.viewerBlockExists(postView.author.did, state)) {
+      return []
+    }
+
+    const childrenByParentUri = this.groupThreadChildrenByParent(
+      anchorUri,
+      uris,
+      state,
+    )
+    const rootUri = getRootUri(anchorUri, post)
+    const opDid = uriToDid(rootUri)
+
+    const anchorTree: ThreadHiddenAnchorPostNode = {
+      item: this.threadHiddenV2ItemPostAnchor({ depth: 0, uri: anchorUri }),
+      replies: this.threadHiddenV2Replies({
+        parentUri: anchorUri,
+        viewer,
+        rootUri,
+        childrenByParentUri,
+        state,
+        below,
+        depth: 1,
+      }),
+    }
+
+    return sortTrimFlattenThreadTree(anchorTree, {
+      opDid,
+      branchingFactor,
+      prioritizeFollowedUsers: false,
+      viewer,
+      fetchedAt: Date.now(),
+    })
+  }
+
+  private threadHiddenV2Replies({
+    parentUri,
+    viewer,
+    rootUri,
+    childrenByParentUri,
+    state,
+    below,
+    depth,
+  }: {
+    parentUri: string
+    viewer: HydrateCtx['viewer']
+    rootUri: string
+    childrenByParentUri: Record<string, string[]>
+    state: HydrationState
+    below: number
+    depth: number
+  }): ThreadHiddenPostNode[] | undefined {
+    // Reached the `below` limit.
+    if (depth > below) {
+      return undefined
+    }
+
+    const childrenUris = childrenByParentUri[parentUri] ?? []
+    return mapDefined(childrenUris, (uri) => {
+      // Not found.
+      const post = state.posts?.get(uri)
+      if (post?.violatesThreadGate) {
+        return undefined
+      }
+      const postView = this.post(uri, state)
+      if (!post || !postView) {
+        return undefined
+      }
+      const authorDid = postView.author.did
+      if (rootUri !== getRootUri(uri, post)) {
+        // outside thread boundary
+        return undefined
+      }
+
+      // Blocked (1p and 3p for replies).
+      const has1pBlock = this.viewerBlockExists(authorDid, state)
+      const has3pBlock =
+        !state.ctx?.include3pBlocks && state.postBlocks?.get(uri)?.parent
+      if (has1pBlock || has3pBlock) {
+        return undefined
+      }
+      if (!this.viewerSeesNeedsReview({ uri, did: authorDid }, state)) {
+        return undefined
+      }
+
+      // No unauthenticated.
+      if (this.noUnauthenticatedPost(viewer, postView)) {
+        return undefined
+      }
```

Comment:
There is a decent amount of common logic needed by both replies and hidden replies.  I wonder if it's worth leaving a note in `threadHiddenV2Replies()` and `threadV2Replies()` to help ensure it stays in sync.